src/types/privatekubectl.ts

```diff
@@ -75,13 +75,17 @@ export class PrivateKubectl extends Kubectl {
             runOutput
          )}`
       )
+
+      if (runOutput.exitCode !== 0) {
```

Comment:
This part looks good to me and makes sense so we don't fail on json parsing like you said.